web/server/codechecker_server/api/report_server.py

```diff
@@ -2903,8 +2903,15 @@ def removeRunReports(self, run_ids, report_filter, cmp_data):
                 LOG.error(ex)
                 return False
 
-        # Remove unused data (files, comments, etc.) from the database.
-        db_cleanup.remove_unused_data(self._Session)
+        # Remove unused comments and unused analysis info from the database.
+        # Originally db_cleanup.remove_unused_data() was used here which
+        # removes unused file entries tool. However removing files at the same
+        # time with a storage concurrently results foreign key constraint
+        # error. An alternative solution can be adding a timestamp to file
+        # entries to delay their removal. The same comment applies to
```

Comment:
...adding the last access timestamp to file entries to delay their removal (and avoid removing frequently accessed files)...